File: php/src/app/dominio/Contexto.php
<?php
namespace dominio;

use dominio\estados;

class Contexto {
    /**
     * @var estados\EstadoContexto
     */
    private $estado;


    public function __construct(){
        $this->estado = new estados\EstadoConcretoA();
    }

    public function actuar(){
        $this->estado->actuar($this);
    }

    public function cambiarEstado(estados\EstadoContexto $estado){
        $this->estado = $estado;
    }
}

File: java/src/main/java/dominio/Contexto.java
package dominio;

import dominio.estados.EstadoConcretoA;
import dominio.estados.EstadoContexto;

public class Contexto {
    private EstadoContexto estado;

    public Contexto(){
        this.estado = new EstadoConcretoA();
    }

    public void actuar(){
        this.estado.actuar(this);
    }

    public void cambiarEstado(EstadoContexto estado){
        this.estado = estado;
    }
}